Project/Project/CDTViewController.m

```diff
@@ -160,13 +167,9 @@ - (void)addTodoButtonTap:(NSObject *)sender {
 -(IBAction)replicateTapped:(id)sender {
     NSLog(@"Replicate");
 
-    dispatch_async(dispatch_get_global_queue(DISPATCH_QUEUE_PRIORITY_DEFAULT, 0), ^{
-        [self.todoReplicator sync];
-        dispatch_async(dispatch_get_main_queue(), ^{
-            [self reloadTasks];
-            [self.tableView reloadData];
-        });
-    });
+    [self.todoReplicator syncInBackgroundWithCompletionHandler:^{
+        [self toggleCompletedShown:nil];
```

Comment:
we shouldn't capture `self` here, we should set a good example, and use:

```objc
__weak CDTViewController *weakSelf = self;
[self.todoReplicator syncInBackgroundWithCompletionHandler:^{
    __strong CDTViewController *self = weakSelf;
    self toggleCompletedShown:nil];
}];
```

or similar.